### ISSUE TITLE
Resource monitoring messages

### ISSUE THREAD

**Issue by danielballan** [ipython/ipykernel#262]
I'm told there is already some work in progress on adding a message type for resource monitoring ('resource_request', 'resource_response') to check on available memory, CPU usage, etc. Following up on an in-person conversation with @ivanov, some thoughts:

* Should this same communication channel provide some static information like `multiprocessing.cpu_count()`? We might draw inspiration from [the informative `_repr_html_` of `dask.distributed.Client`](https://github.com/dask/distributed/blob/master/distributed/client.py#L582).
* It would be cool to support a notebook / JupyterLab extension to display (and actively update) visual indicators of memory and CPU usage on the machine where the kernel is running.

I'd be happy to write some code for this, but I don't want to step on any existing branches. If these comments would fit into an existing GH issue, feel free to close this one and direct me there.

**Comment by ivanov** [comment]
Yes, this is being tracked over at jupyter/jupyter#264, since it will involve changes in the messaging spec, the kernel, the frontends. The current thinking is that yes, mime bundles will be the way to go, since the kernel won't know what the capabilities of the client will be.
